File: 66. Plus One/py/plusOne.py
# LeetCode problem: 66. Plus One
# <https://leetcode.com/problems/plus-one/>

from typing import List


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        src_number = 0
        exp_factor = 0
        for d in reversed(digits):
            src_number += d * 10 ** exp_factor
            exp_factor += 1
        dst_number = src_number + 1
        dst_list = []
        for char in str(dst_number):
            dst_list.append(int(char))
        return dst_list

File: 66. Plus One/cs/PlusOne/PlusOne.cs
// LeetCode problem: 66. Plus One
// https://leetcode.com/problems/plus-one/

namespace LeetCode.PlusOne;

public class Solution
{
    public int[] PlusOne(int[] digits)
    {
        for (int i = digits.Length - 1; i >= 0; i--)
        {
            if (digits[i] < 9)
            {
                digits[i] += 1;
                return digits;
            }
            digits[i] = 0;
        }
        return [1, ..digits];
    }
}